fsd_utils/configclass.py

```diff
@@ -0,0 +1,50 @@
+def configclass(cls):
+
+    # Checks base classes for _config_info_, a dict containing
+    # all infomation about config. attrs, and
+    # adds them to a list.
+    config_infos = [
+        base.__dict__["_config_info_"]
+        for base in cls.__bases__
+        if "_config_info_" in base.__dict__
+    ]
+
+    # We compile all of these _config_info_'s into a dictionary,
+    #  with overwritten values.
+    settled_bases_config_info = {
+        k: v for config_dict in config_infos for k, v in config_dict.items()
+    }
+
+    # We extract the values from the class and overlay
+    #  the above dict with its values.
+    for k, v in cls.__dict__.items():
+        if not (k.startswith("__") or k.endswith("__")):
+            settled_bases_config_info[k] = v
+
+    # This is the dicitonary that will be used
+    #  by classes which inherit this class
+    cls._config_info_ = settled_bases_config_info
+
+    @classmethod
+    def pretty_print(self):
```

Comment:
In future we should allow this to work with the logging utils so any pretty printing is written to logs not console

Reply:
Sounds good :) I'll create a BAU pr once harry's stuff has been added.